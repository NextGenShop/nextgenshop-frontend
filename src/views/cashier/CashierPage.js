import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Basket from "../../components/Basket";
import VideoChat from "../../components/VideoChat";
import ProductCatalog from "../../components/ProductCatalog";
import mockBasketItems from "../../store/mockdata/MockBasketItems.json";
import ProductSearchBox from "../../components/ProductSearchBox";
import { addBasketItem, removeBasketItem } from "../../utils/basketUtils";

const useStyles = makeStyles((theme) => ({
  heading: {
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
  },
  media: {
    height: 0,
    paddingTop: "56.25%", // 16:9
  },
  mr: {
    marginRight: theme.spacing(1),
  },
  mt: {
    marginTop: theme.spacing(3),
  },
  divider: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
  },
}));

export default function CashierPage() {
  const classes = useStyles();
  const [searchQuery, setSearchQuery] = React.useState("");
  const [items, setItems] = React.useState(mockBasketItems);

  const mockRetailer = "Waitrose";

  return (
    <Container maxWidth="xl">
      <Typography variant="h2" className={classes.heading}>
        NextGenShop Cashier
      </Typography>
      <Grid container spacing={3}>
        <Grid item md={6}>
          <Typography variant="subtitle1" display="inline" className={classes.mr}>
            Video Call
          </Typography>
          <Typography variant="subtitle2" display="inline">
            {" "}
            Customer Name
          </Typography>
        </Grid>
        <Grid item xs={6}>
          <Grid container spacing={3}>
            <Grid item xs={12}>
              <Typography variant="subtitle1">Product Catalogue</Typography>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item md={6}>
          <VideoChat />
        </Grid>
        <Grid item xs={6}>
          <Grid container spacing={3}>
            <Grid item xs={12}>
              <ProductSearchBox
                value={searchQuery}
                onChange={(e) => {
                  setSearchQuery(e.target.value);
                }}
              />
              <ProductCatalog
                tableView
                retailer={mockRetailer}
                addToBasket={(product) => {
                  addBasketItem(product, items, setItems);
                }}
                searchQuery={searchQuery}
              />
              <hr className={classes.mt} />
              <Typography variant="subtitle1">Customer Basket</Typography>
              <Basket
                items={items}
                removeItem={(basketItemId) => removeBasketItem(basketItemId, items, setItems)}
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Container>
  );
}
