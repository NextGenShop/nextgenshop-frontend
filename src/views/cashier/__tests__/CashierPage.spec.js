import React from 'react';
import { shallow } from 'enzyme';
import CashierPage from '../CashierPage';

describe('CashierPage', () => {
  test('should correctly render', () => {
    const component = shallow(<CashierPage />);
    expect(component).toMatchSnapshot();
  });
});
