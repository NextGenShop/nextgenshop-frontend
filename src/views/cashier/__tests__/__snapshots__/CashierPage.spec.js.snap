// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CashierPage should correctly render 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-root-1"
  maxWidth="xl"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-mr-3"
        display="inline"
        variant="subtitle1"
      >
        Video Call
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        display="inline"
        variant="subtitle2"
      >
         
        Customer Name
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={6}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            variant="subtitle1"
          >
            Product Catalogue
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
    >
      <VideoChat />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={6}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <ProductSearchBox
            onChange={[Function]}
            value=""
          />
          <Connect(ProductCatalog)
            retailer="Mock Retailer"
            searchQuery=""
            shopperId={0}
            tableView={true}
          />
          <hr
            className="makeStyles-mt-4"
          />
          <WithStyles(ForwardRef(Typography))
            variant="subtitle1"
          >
            Customer Basket
          </WithStyles(ForwardRef(Typography))>
          <Connect(Basket)
            shopperId={0}
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
